sdk/testing/stepwise/environments/docker/environment.go

```diff
@@ -0,0 +1,858 @@
+	transport := cleanhttp.DefaultPooledTransport()
+	transport.TLSClientConfig = n.TLSConfig.Clone()
+	if err := http2.ConfigureTransport(transport); err != nil {
+		return nil, err
+	}
+	client := &http.Client{
+		Transport: transport,
+		CheckRedirect: func(*http.Request, []*http.Request) error {
+			// This can of course be overridden per-test by using its own client
+			return fmt.Errorf("redirects not allowed in these tests")
+		},
+	}
+	config := api.DefaultConfig()
+	if config.Error != nil {
+		return nil, config.Error
+	}
+	config.Address = fmt.Sprintf("https://127.0.0.1:%s", n.HostPort)
+	config.HttpClient = client
+	config.MaxRetries = 0
+	apiClient, err := api.NewClient(config)
+	if err != nil {
+		return nil, err
+	}
+	apiClient.SetToken(n.Cluster.RootToken())
+	return apiClient, nil
+}
+
+// Cleanup kills the container of the node
+func (n *DockerClusterNode) Cleanup() error {
+	return n.dockerAPI.ContainerKill(context.Background(), n.container.ID, "KILL")
+}
+
+func (n *DockerClusterNode) Start(cli *docker.Client, caDir, netName string, netCIDR *DockerClusterNode, pluginBinPath string) error {
+	n.dockerAPI = cli
+
+	err := n.setupCert()
+	if err != nil {
+		return err
+	}
+
+	vaultCfg := map[string]interface{}{
+		"listener": map[string]interface{}{
+			"tcp": map[string]interface{}{
+				"address":       fmt.Sprintf("%s:%d", "0.0.0.0", 8200),
+				"tls_cert_file": "/vault/config/cert.pem",
+				"tls_key_file":  "/vault/config/key.pem",
+				"telemetry": map[string]interface{}{
+					"unauthenticated_metrics_access": true,
+				},
+			},
+		},
+		"telemetry": map[string]interface{}{
+			"disable_hostname": true,
+		},
+		"storage": map[string]interface{}{
+			"raft": map[string]interface{}{
+				"path":    "/vault/file",
+				"node_id": n.NodeID,
+			},
+		},
+		"cluster_name":         netName,
+		"log_level":            "TRACE",
+		"raw_storage_endpoint": true,
+		"plugin_directory":     "/vault/config",
+		// disable_mlock is required for working in the Docker environment with
+		// custom plugins
+		"disable_mlock": true,
+		// These are being provided by docker-entrypoint now, since we don't know
+		// the address before the container starts.
+		//"api_addr": fmt.Sprintf("https://%s:%d", n.Address.IP, n.Address.Port),
+		//"cluster_addr": fmt.Sprintf("https://%s:%d", n.Address.IP, n.Address.Port+1),
+	}
+	cfgJSON, err := json.Marshal(vaultCfg)
+	if err != nil {
+		return err
+	}
+
+	err = ioutil.WriteFile(filepath.Join(n.WorkDir, "local.json"), cfgJSON, 0644)
+	if err != nil {
+		return err
+	}
+	// setup plugin bin copy if needed
+	copyFromTo := map[string]string{
+		n.WorkDir: "/vault/config",
+		caDir:     "/usr/local/share/ca-certificates/",
+	}
+	if pluginBinPath != "" {
+		base := path.Base(pluginBinPath)
+		copyFromTo[pluginBinPath] = filepath.Join("/vault/config", base)
+	}
+
+	r := &Runner{
+		dockerAPI: cli,
+		ContainerConfig: &container.Config{
+			Image: "vault",
```

Comment:
This will default to `latest`.  I think we might need the ability to specify which version of the Docker image to use.  Thoughts?  For example, these tests on a release branch would run on a newer version of Vault potentially.

Reply:
Configuring the version of Vault is on the roadmap, but intentionally deferred in the MVP

Reply:
If this isn't configurable and only points to `vault:latest`, which pulls this image in from Dockerhub, we can't run acceptance tests against the version we're trying to release.  Maybe that's by design but I think it could give falser positive results with an older core.  

For example, in the upcoming 1.5 release, there is no 1.5 image in Dockerhub, so how do we utilize these acceptance tests for that version?